pkg/kafka/policy.go

```diff
@@ -141,15 +141,14 @@ func matchOffsetFetchReq(req *proto.OffsetFetchReq, rule api.PortRuleKafka) bool
 	return true
 }
 
-func (req *RequestMessage) ruleMatches(rule api.PortRuleKafka) bool {
+func (req *RequestMessage) RuleMatches(rule *api.PortRule) bool {
 	if req == nil {
 		return false
 	}
 
-	flowdebug.Log(log.WithFields(logrus.Fields{
-		fieldRequest: req.String(),
-		fieldRule:    rule,
-	}), "Matching Kafka rule")
+	if flowdebug.Enabled() {
+		log.Debugf("Matching Kafka request %s against rule %v", req.String(), *rule)
```

Comment:
Why the change to logging method? Seems like structured log was nicer, would it be possible to get back to it?

Reply:
Short answer: to avoid confusing log messages.

Long answer: Kafka logs become part of Envoy logs when Kafka is moved to the proxylib. Envoy log messages are not structured and currently we have no code to try parse the structured items from the main log message. This means that if we use structured logs here, they will be embedded in to the main log message and will not be found by any code that will try to parse structured logs. For reference please see https://github.com/cilium/cilium/blob/master/pkg/envoy/envoy.go#L282-L343